### PR TITLE
Correctly check inaxes for multicursor

### PR DESCRIPTION
**Pull request by ianhi**
## PR Summary
multicursor now checks if `event.inaxes` is one of the axes it will draw on.

Fixes: https://github.com/matplotlib/matplotlib/issues/19635

Before
Peek 2021-03-03 18-59
After:

Peek 2021-03-03 19-28

## PR Checklist

There aren't any multicursor tests - does this change need a test?

- [ NA? ] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [NA] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [x] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [NA] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [NA] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

**Comment by QuLogic** [comment]
There are very few widget tests, so if you want to write one, that'd be great.

File: lib/matplotlib/widgets.py [hunk 1]
```diff
@@ -1653,7 +1653,7 @@ def clear(self, event):
     def onmove(self, event):
         if self.ignore(event):
             return
-        if event.inaxes is None:
+        if not event.inaxes in self.axes:
```
Comment (dstansby) [inline lib/matplotlib/widgets.py:1656]:
```suggestion
        if event.inaxes not in self.axes:
```
This should make the flake checker happy

**Review by dstansby — CHANGES_REQUESTED** [review]
Thanks a lot for this! I've commented on what needs to change for the tests to pass. In addition, do you think you'll be able to add a new test to check for the bug that you've fixed? If not, I can have a go at adding one.

**Comment by ianhi** [comment]
> do you think you'll be able to add a new test to check for the bug that you've fixed? If not, I can have a go at adding one.

Hi @dstansby thanks for pinging me here. Work got busy, and it suddenty wasn't allllways below freezing so I've been making a concsious effort to be outside instead of on my computer and let this slip - sorry about that! Anyway I added a test which I think covers most what seems relevant for MultiCursor.

File: lib/matplotlib/widgets.py (hunk 1 above)
Comment (ianhi) [reply lib/matplotlib/widgets.py]: indeed. thanks!

**Review by dstansby — APPROVED** [review]
Thanks a lot!

**Review by timhoffm — APPROVED** [review]

**Comment by ianhi** [comment]
Since this is a bugfix could it be re-milestoned to `3.4.x` and then caught up in a patch release?

Apparently people are running into this, see https://github.com/matplotlib/matplotlib/issues/20231

**Comment by tacaswell** [comment]
@meeseeksdev backport to v3.4.x

**Pull request merged**